src/lib/containers/AccessApprovalsTable.tsx

```diff
@@ -0,0 +1,213 @@
+import moment from 'moment'
+import React, { useEffect, useState } from 'react'
+import { Table } from 'react-bootstrap'
+import { SMALL_USER_CARD } from '../utils/SynapseConstants'
+import { useSynapseContext } from '../utils/SynapseContext'
+import {
+  AccessApprovalSearchRequest,
+  AccessApprovalSearchResponse,
+  AccessApprovalSearchResult,
+} from '../utils/synapseTypes'
+import { accessApprovalSearch } from '../utils/SynapseClient'
+import Typography from '../utils/typography/Typography'
+import IconSvg from './IconSvg'
+import UserCard from './UserCard'
+import { displayToast } from './ToastMessage'
+import { PRODUCTION_ENDPOINT_CONFIG } from '../utils/functions/getEndpoint'
+import { SynapseSpinner } from './LoadingScreen'
+export type AccessApprovalsTableProps = {
+  accessorId: string
+  accessRequirementId?: string
+}
+export const AccessApprovalsTable: React.FunctionComponent<
+  AccessApprovalsTableProps
+> = ({ accessorId, accessRequirementId }) => {
+  const { accessToken } = useSynapseContext()
+  const [searchResult, setSearchResult] =
+    useState<AccessApprovalSearchResponse>()
+  const [isLoading, setIsLoading] = useState(false)
+  const [arList, setArList] = useState<AccessApprovalSearchResult[]>()
+
+  const getApprovals = async () => {
+    try {
+      const searchRequest: AccessApprovalSearchRequest = {
+        accessorId,
+        accessRequirementId,
+      }
+      const approvalResponse = await accessApprovalSearch(
+        searchRequest,
+        accessToken,
+      )
+      setSearchResult(approvalResponse)
+      setArList(approvalResponse.results)
+    } catch (err: any) {
+      displayToast(err.reason as string, 'danger')
+    }
+  }
+
+  const onLoadMore = async () => {
+    setIsLoading(true)
+    try {
+      const searchRequest: AccessApprovalSearchRequest = {
+        accessorId,
+        accessRequirementId,
+        nextPageToken: searchResult?.nextPageToken,
+      }
+      const approvalResponse = await accessApprovalSearch(
+        searchRequest,
+        accessToken,
+      )
+      setSearchResult(approvalResponse)
+      if (arList) {
+        setArList([...arList, ...approvalResponse.results])
+      }
+    } catch (err: any) {
+      displayToast(err.reason as string, 'danger')
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  useEffect(() => {
+    getApprovals()
+  }, [accessToken])
```

Comment:
This will only run when the access token changes, so new data won't be fetched e.g. if accessorId changes.

ESLint should catch this if you define the function within the effect.